File: Dasar-dasar Pemrograman/Semester 1/JavaScript/Rekursi/faktorialrec.js
const readline = require("readline");
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function faktorial (param) {
    if (param == 2) {
        return 2;
    }

    if (param > 2) {
        return param * faktorial(param - 1);
    }

    return 1;
}

rl.question("Masukkan nilai: ", function(nilai) {
    let faktor = faktorial(nilai);
    console.log(faktor);
    rl.close();
});

rl.on("close", function() {
    process.exit(0);
});

File: Dasar-dasar Pemrograman/Semester 1/Python/Rekursi/faktorialrec.py
param = input("Masukkan nilai: ")

def faktorial(param):
    if param == '':
        return None

    parameter = int(param)
    if parameter > 2:
        return parameter * faktorial(parameter - 1)

    if parameter == 2:    
        return 2
    
    return 1

faktor = faktorial(param)
print(faktor)
